src/OracleWhitelist.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {UniswapV3Oracle} from "./base/UniswapV3Oracle.sol";
+import {Initializable} from "./base/Initializable.sol";
+import {IOracleWhitelist} from "./interfaces/IOracleWhitelist.sol";
+
+/**
+ * @title The contract handles whitelist related features
+ * @notice The main functionalities are:
+ * - Self whitelist by sending ETH to this contract(only when self whitelist is allowed - controlled by _isSelfWhitelistDisabled flag)
+ * - Ownable: Add whitelisted addresses
+ * - Ownable: Set max ETH amount to buy(default 3 ETH)
+ * - Ownable: Set univ3 TWAP oracle
+ * - Token contract `_beforeTokenTransfer` hook will call `checkWhitelist` function and this function will check if buyer is eligible
+ */
+contract OracleWhitelistWhitelist is IOracleWhitelist, UniswapV3Oracle, Ownable, Initializable {
+    /// @dev Maximum ETH amount to contribute
+    uint256 private _maxAddressCap;
+    /// @dev Flag for locked period
+    bool private _locked;
+    /// @dev Token token contract address
+    uint256 private _whitelistCount;
+    /// @dev Max index allowed
+    uint256 private _allowedWhitelistIndex;
```

Comment:
We don't need this either.

Probably could store the whitelist as of true/false, or store in in an array?

Reply:
I'm trying to keep vultisig's logic as much as possible